### PR TITLE
docs(react): update react tutorial to match the latest version of Nx

### LINKED ISSUE #7638: Tutorials need to be cleaned up to match the latest release
## Current Behavior

Some examples:
1. Tutorials are still referring to the old format of workspace.json
2. proxy.conf.json is not found where the tutorial is describing

## Expected Behavior

Tutorial should be in sync with what Nx is actually doing.

## Steps to Reproduce

Just work through the tutorial step-by-step.

### PR DESCRIPTION
**Pull request by nareshbhatia**
## Current Behavior

React tutorial instructions do not match the actual behavior of Nx.

## Expected Behavior

React tutorial instructions should match the actual behavior of Nx.

## Related Issue(s)

Fixes #7638

### CODE REVIEW

File: docs/react/tutorial/01-create-application.md [hunk 1]
```diff
@@ -23,11 +23,12 @@ Workspace name (e.g., org name)     myorg
 What to create in the new workspace react
 Application name                    todos
 Default stylesheet format           CSS
+Use Nx Cloud? (It's free and doesn't require registration.) No
```
Comment (brandonroberts) [inline docs/react/tutorial/01-create-application.md:26]: I would remove this and leave the choice up to the person going through the tutorial

File: docs/react/tutorial/05-add-node-app.md [hunk 2]
```diff
@@ -147,3 +150,7 @@ const server = app.listen(port, () => {
 });
 server.on('error', console.error);
 ```
+
+**Now run `npx nx serve api` to run the api server**
+
+Finally, refresh the front-end to hit the server. You will see that the React app is now able to fetch and create todos by calling the API.
```
Comment (brandonroberts) [inline docs/react/tutorial/05-add-node-app.md:156]:
```suggestion
Refresh the application in the browser. The React app is now able to fetch and create todos by calling the API.
```

File: docs/react/tutorial/06-proxy.md [hunk 3]
```diff
@@ -40,21 +40,21 @@ It created a proxy configuration that allows the React application to talk to th
...
-Every target uses a builder which actually runs this target. So targets are analogous to typed npm scripts, and builders are analogous to typed shell scripts.
+Every target uses an executor which actually runs this target. So targets are analogous to typed npm scripts, and executors are analogous to typed shell scripts.
 
 **Why not use shell scripts and npm scripts directly?**
 
 There are a lot of advantages to providing additional metadata to the build tool. For instance, you can introspect targets. `npx nx serve todos --help` results in:
 
 ```bash
-npx nx run todos:serve [options,...]
+nx run todos:serve [options,...]
```
Comment (brandonroberts) [inline docs/react/tutorial/06-proxy.md:54]: Keep as-is

```suggestion
npx nx run todos:serve [options,...]
```

File: docs/react/tutorial/08-create-libs.md [hunk 4]
```diff
@@ -87,21 +86,20 @@ myorg/
...
+│       │   │   └── ui.tsx
 │       │   └── index.ts
 │       ├── jest.conf.js
-│       ├── tsconfig.app.json
+│       ├── project.json
 │       ├── tsconfig.json
+│       ├── tsconfig.lib.json
 │       └── tsconfig.spec.json
 ├── tools/
-├── workspace.json
```
Comment (brandonroberts) [inline docs/react/tutorial/08-create-libs.md:104]: Why was this removed?

File: docs/react/tutorial/09-dep-graph.md [hunk 5]
```diff
@@ -7,3 +7,11 @@ An Nx workspace can contain dozens or hundreds of applications and libraries. As
...
 
 With Nx, you can do better than that.
+
+Run
+
+```bash
+npx nx dep-graph
+```
+
+The Dependency Graph page should load in your browser. Click on "Show all projects" to show a dependency graph all the apps and libraries in the workspace.
```
Comment (brandonroberts) [inline docs/react/tutorial/09-dep-graph.md:17]:
```suggestion
The dependency graph page opens in a new browser window. Click on "Show all projects" to see all the apps and libraries in the workspace.
```

**Review by brandonroberts — DISMISSED** [review]

**Comment by brandonroberts** [comment]
Thanks for the PR @nareshbhatia! I left some feedback

File: docs/react/tutorial/08-create-libs.md (hunk 4 above)
Comment (nareshbhatia) [reply docs/react/tutorial/08-create-libs.md:104]: It was not there in the previous treeview on the same page. So I decided to keep it consistent here. But I will add it to both places now.

**Comment by nareshbhatia** [comment]
@brandonroberts, thanks for the feedback. I have implemented all the suggestions and pushed a new commit.

**Comment by nareshbhatia** [comment]
@brandonroberts, not sure if I have done this the right way, but I took all your suggestions and pushed them to the PR through my fork. I saw that you had made a few suggestions that I could commit directly, but wasn't sure how that worked.

**Comment by nareshbhatia** [comment]
Ah, I see now! You used the "Insert a suggestion" feature in GitHub. Very nice!

**Comment by brandonroberts** [comment]
@nareshbhatia either is fine! Will you rebase your branch on master and push it up so it can run the CI checks? 
Thanks!

**Comment by nareshbhatia** [comment]
@brandonroberts, I have rebased on master and pushed.

**Comment by nareshbhatia** [comment]
@brandonroberts, I pushed 12 hours ago, still it says "Some checks haven’t completed yet". Anything left for me to do?

**Comment by nareshbhatia** [comment]
@brandonroberts, @FrozenPandaz - what is the update on this? First class up-to-date docs should be a priority, no?

**Comment by brandonroberts** [comment]
@nareshbhatia yep. There are some failed CI checks. Looks like you need to format the docs and push up again.

```sh
yarn nx format
```

After you commit your changes, run `yarn documentation` to make sure. Then push up

**Comment by nareshbhatia** [comment]
@brandonroberts, I ran `yarn format` and `yarn documentation`. All looks good now. Rebased and pushed to my branch.

**Review by FrozenPandaz — APPROVED** [review]
Thank you for your contribution! LGTM :tada:

**Comment by nareshbhatia** [comment]
@brandonroberts, @FrozenPandaz, this changes are not showing up in the live site yet. When does the site gets published?

**Comment by brandonroberts** [comment]
The site is published on every green commit on master, so it should be there. We'll take a look
